pkg/tracing/otlp_sdk.go

```diff
@@ -0,0 +1,119 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"time"
+
+	"go.opentelemetry.io/contrib/exporters/autoexport"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.25.0"
+
+	errors "github.com/zgalor/weberr"
+)
+
+// ConfigureOpenTelemetryTracer configures the global OpenTelemetry trace
+// provider.
+//
+// The function uses the following environment variables for the tracer
+// configuration:
+//   - `OTEL_TRACES_EXPORTER`, either `otlp` to send traces to an OTLP endpoint or `console`.
+//   - `OTEL_EXPORTER_OTLP_TRACES_PROTOCOL`, either `grpc` or `http`.
+//   - `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT`, endpoint where to send the OTLP
+//     traces (e.g. `https://localhost:4318/v1/traces`).
+//
+// See
+// https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp
+// for the list of all supported variables.
+//
+// An error is returned if an environment value is set to an unhandled value.
+//
+// If no environment variable are set, a no-op tracer is setup.
+func ConfigureOpenTelemetryTracer(ctx context.Context, logger *slog.Logger, serviceName string, serviceVersion string, cloudProvider string, resourceAttrs ...attribute.KeyValue) (func(context.Context) error, error) {
```

Comment:
There is a concern about what kind of logger we use here. Different components seem to use different loggers. That makes this hard to be generic on loggger parameter.

Reply:
I would even go with a simple `func(err error)`. The information about whether or not the exporter is a no-op can be returned to the caller. E.g.

```
func ConfigureOpenTelemetryTracer(
    ctx context.Context,
    errCallback func(error),
    serviceName string,
    serviceVersion string,
    cloudProvider string,
    resourceAttrs ...attribute.KeyValue) (func(context.Context) error, bool, error) {
```

Reply:
As we discussed on Thursday, the `logr` usage is the best option to avoid wiriting yet-another logging interface. `logr` is a logging library + a interface that can be adopted. It is also currently being used by other popular logging libraries. It suits our needs well.
I also removed some specific attributes to simplify the func signatures. User can add those custom attributes to input slice if needed.